### PR TITLE
Updated hostnamectl completions

### PR DESCRIPTION
**Pull request by exploide**
## Description

`hostnamectl` has been changed some time ago. The subcommands have different names and there is a bit of new functionality. I updated the completions to conform with the new behavior.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

**Comment by mqudsi** [comment]
So these look like they're breaking changes. Do you know what systems have the old version and what systems have the new version? This ships with the distro and isn't updated out-of-stream, so we can't just assume users have the latest version.

**Review by mqudsi — CHANGES_REQUESTED** [review]
So these look like they're breaking changes. Do you know what systems have the old version and what systems have the new version? This ships with the distro and isn't updated out-of-stream, so we can't just assume users have the latest version.

File: share/completions/hostnamectl.fish [hunk 1]
```diff
@@ -1,18 +1,21 @@
-set -l commands status set-hostname set-icon-name set-chassis set-deployment set-location
```
Comment (EmilySeville7cfg) [inline share/completions/hostnamectl.fish:1]: Optional fix: if `--version` or option with the same semantics is available please use it to decide what completions to provide.

**Comment by exploide** [comment]
Made the subcommand names conditional based on hostnamectl version.

**Comment by mqudsi** [comment]
Thanks, this looks good.

**Review by mqudsi — APPROVED** [review]

**Pull request merged**